src/core/index_meta.rs

```diff
@@ -112,10 +112,16 @@ impl SegmentMeta {
             .load(std::sync::atomic::Ordering::Relaxed)
```

Comment:
instead of `list_files`, it could make sense to have `list_active_files` and `list_all_files`